packages/core/useFetch/index.ts

```diff
@@ -386,13 +439,22 @@ export function useFetch<T>(url: MaybeRef<string>, ...args: any[]): UseFetchRetu
 
   function setMethod(method: string) {
     return (payload?: unknown, payloadType?: PayloadType) => {
-      if (!initialized) {
+      if (!initialized)
```

Comment:
I don't know if this PR will be merged to master (if this has no breaking changes) or to `next` branch.
I prefer the check there https://github.com/vueuse/vueuse/blob/338b4a678f071a25eb76d77345e651e7ff5c4b1f/packages/core/useFetch/index.ts#L396. 

From my use case I have a component that first get some data, and if the user makes changes it `put` to the same url (simple REST), Using the `next` branch is really convenient creating only one `useFetch`
```js
const {isFetching, data, put, error} = useFetch('/resource').json().get()

// elsewhere
put({/*updated data*/}).execute()
```
the same `isFetching` or `error` are used.

Here if more `put`, `post`, `delete` are called it will use the first method registered and it will be confusing.

Reply:
We are waiting for @antfu review, I think changing method will not be a problem (if well documented and including a breaking change like throwing an error while fetching...), your use case sounds reasonable, and including another `useFetch` is a pain, just see demo page in this PR. 

I'm awaiting for Jacob to see if I include here to provide the payload on error. I can confirm that it is necessary, since the user can use an error with payload: for example, I send a 413 when the size exceeds the server max size, and in general I always send a JSON payload from the server.

I suggest this changes (maybe we can be more concise, suach as including `responseError?: boolean | undefined`):

![imagen](https://user-images.githubusercontent.com/6311119/120542602-93d6f080-c3eb-11eb-9cef-8b2157d395a7.png)

![imagen](https://user-images.githubusercontent.com/6311119/120544239-7c990280-c3ed-11eb-8370-ac414ea631df.png)

![imagen](https://user-images.githubusercontent.com/6311119/120543776-fd0b3380-c3ec-11eb-9f39-bb21b9f259f0.png)

Reply:
In fact I think there is a problem to directly consuming the response body: for example, downloading a file can result in 403 or 401 or 404, that is, unauthorized , permission denied or not found, but there is no blob to download.

The user must have the control via the callback, and its responsability will be to extract the response body after check if there is an error.

So a second change will be to delegate the responsability to the user when the `afterFetch` is provided.

Reply:
Maybe this will be necessary to be moved to a dicussion.

Reply:
One last thing, if the server is down or the client is offline, the response will be null or undefined so we need handle this case, maybe using `useOffline` before executing the request...

The browser will throw a generic error, I handle the situation using literals:
`failed to fetch` and `networkerror when attempting to fetch resource.`